server/handlers.go

```diff
@@ -668,6 +668,54 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 			s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
 			return
 		}
+
```

Comment:
If anything fails here, you'll need to clean up that refresh token you just created.

Maybe a `defer`?

```
defer func() {
    if !createdOfflineSession {
        // delete refresh token.
    }
}()
```